src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
 
 import { User } from "types/User";
 
-// @debt remove styled-components in favour of using our standard scss patterns
-import * as S from "./UserProfilePicture.styles";
 import "./UserProfilePicture.scss";
 
+const randomAvatarUrl = (id: string) =>
+  "/avatars/" +
+  RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
+
+// TODO: refactor this to accept named props instead of positional args
+const avatarUrl = (
+  id: string,
+  anonMode?: boolean,
+  pictureUrl?: string,
+  miniAvatars?: boolean
+): string => {
+  if (anonMode || !id) {
+    return DEFAULT_PROFILE_IMAGE;
+  }
+
+  if (!miniAvatars && pictureUrl) {
+    return pictureUrl;
+  }
+
+  // TODO: how is this intended to be used? Why doesn't it use the profile image? It seems to be something set on venue config..
+  if (miniAvatars) {
+    return randomAvatarUrl(id);
+  }
+
+  return DEFAULT_PROFILE_IMAGE;
+};
+
 export interface UserProfilePictureProp {
   user: WithId<User>;
   isAudioEffectDisabled?: boolean;
   miniAvatars?: boolean;
   avatarClassName?: string;
   avatarStyle?: object;
   containerStyle?: object;
-  reactionPosition?: "right" | "left" | undefined;
+  reactionPosition?: "left" | "right";
 }
 
 // @debt This component should be divided into a few with simpler logic. Also, remove `styled components`
 // @debt the UserAvatar component serves a very similar purpose to this, we should unify them as much as possible
```

Comment:
[TODO]
- This component should be divided into a few with simpler logic. Also, remove `styled components`
- the UserAvatar component serves a very similar purpose to this, we should unify them as much as possible

Reply:
Cleaned up comments in https://github.com/sparkletown/sparkle/pull/1323/commits/e7d9b0644fa1364a170b627dae905206a6da7121